examples/classification_example/main_train.py

```diff
@@ -0,0 +1,238 @@
+    word = batch["text_tag"]["data"]
+    labels = batch["label_tag"]["data"]
+    optim.zero_grad()
+
+    logits, pred = None, None
+
+    if config.config_model.model == "cnn":
+        logits, pred = model(batch)
+
+    if config.config_model.model == "bert":
+        logits, pred = model(pad_each_bach(word, max_sen_length))
+
+    labels_1D = torch.squeeze(labels)
+    torch.add(labels_1D, -2, out=labels_1D)
+
+    loss = criterion(logits, labels_1D)
+
+    loss.backward()
+    optim.step()
+
+    batch_train_err = loss.item() * batch.batch_size
+
+    return batch_train_err
+
+
+# All the configs
+config_data = yaml.safe_load(open("config_data.yml", "r"))
+config_model = yaml.safe_load(open("config_model.yml", "r"))
+
+config = Config({}, default_hparams=None)
+config.add_hparam('config_data', config_data)
+config.add_hparam('config_model', config_model)
+
+tp_request = {
+    "scope": Sentence,
+    "schemes": {
+        "text_tag": {
+            "entry_type": Token,
+            "attribute_get": "text",
+            "vocab_method": "indexing",
+            "type": TrainPreprocessor.DATA_INPUT,
+            "extractor": AttributeExtractor
+        },
+        "label_tag": {
+            "entry_type": Sentence,
+            "attribute_get": "speaker",
+            "vocab_method": "indexing",
+            "type": TrainPreprocessor.DATA_OUTPUT,
+            "extractor": AttributeExtractor
+        }
+    }
+}
+
+# All not specified dataset parameters are set by default in Texar.
+# Default settings can be found here:
+# https://texar-pytorch.readthedocs.io/en/latest/code/data.html#texar.torch.data.DatasetBase.default_hparams
+tp_config = {
+    "preprocess": {
+        "device": device.type
+    },
+    "dataset": {
+        "batch_size": config.config_data.batch_size_tokens
+    }
+}
+
+processor_config = {
+    'augment_entry': "ft.onto.base_ontology.Token",
+    'other_entry_policy': {
+        "kwargs": {
+            "ft.onto.base_ontology.Sentence": "auto_align"
+        }
+    },
+    'type': 'data_augmentation_op',
+    'data_aug_op': 'tests.forte.processors.base.data_augment_replacement_processor_test.TmpReplacer',
+    "data_aug_op_config": {
+        'kwargs': {}
+    },
+    'augment_pack_names': {
+        'kwargs': {}
+    }
+}
+
+imdb_train_reader = IMDBReader(cache_in_memory=True)
+
+pl = Pipeline()
+pl.set_reader(imdb_train_reader)
+pl.add(ReplacementDataAugmentProcessor(), processor_config)
+pl.initialize()
+
+datapack_generator = pl.process_dataset(config.config_data.train_path)
+
+
+train_preprocessor = TrainPreprocessor(pack_generator=datapack_generator,
+                                       request=tp_request,
+                                       config=tp_config)
+
+
+max_sen_length = 0
+train_batch_iter: Iterator[Batch] = \
+        train_preprocessor.get_train_batch_iterator()
+
+# Fing the max sentence length in the whole dataset
+for batch in tqdm(train_batch_iter):
```

Comment:
Normally we don't use the max_length, since it can be too large. How about we just manually set a number here.

Reply:
I did this in prediction. In the train, I still think run through the main train to find the max_length over dataset is necessary for CNN model.